### ISSUE TITLE
slower queries?

### ISSUE THREAD

**Issue by jrmlhermitte** [bluesky/databroker#299]
This might be just me, but I figure the more documentation the better so I'll write it.

A query like this:
```python
hdrs = cmsdb(start_time="2017-06-01", stop_time="2017-07-01")
next(iter(hdrs))
```
returns a result near instantaneously.
However, a restreaming as such:
```python
hdrs = cmsdb(start_time="2017-06-01", stop_time="2017-07-01")
stream = cmsdb.restream(hdrs, fill=False)
nds = next(iter(stream))
```
hangs for a long time (5 min or more).

Any idea why this may be happening?

I'm using databroker version 0.9.0+16.gd78bbfa
The broker object is constructed as follows:
```python
from databroker.broker import Broker
from databroker.headersource.mongo import MDSRO
from databroker.assets.mongo import RegistryRO

    mds_conf = {
                'host': host,
                'port': port,
                'database': mdsname,
                'timezone': 'US/Eastern',
                 }

    reg_conf = {
            'host': host,
            'port': port,
            'database': fsname
    }

    mds = MDSRO(mds_conf)
    reg = RegistryRO(reg_conf)
    db = Broker(mds, reg)
```

**Comment by danielballan** [comment]
It looks like `check_fields_exist` is applied to all the headers in a separate loop, breaking the laziness of restream.

**Comment by danielballan** [comment]
Fixed in #300 and merged in time for v0.9.1.
